src/analysis/locate.ml

```diff
@@ -809,6 +809,33 @@ let from_longident ~config ~env ~lazy_trie ~pos nss ml_or_mli ident =
     else
       locate ~config ~ml_or_mli ~path:tagged_path ~lazy_trie ~pos ~str_ident loc
 
+let from_path ~config ~local_defs ~pos ~namespace ml_or_mli path =
+  let str_ident = Path.name path in
+  let browse = Mbrowse.of_typedtree local_defs in
+  let lazy_trie =
+    lazy (Typedtrie.of_browses ~local_buffer:true
+            [Browse_tree.of_browse browse])
+  in
+  let nss_path = Namespaced_path.of_path ~namespace path in
+  let loc = Location.none in
+  match
+    locate ~config ~ml_or_mli ~path:nss_path ~lazy_trie ~pos ~str_ident loc
+  with
```

Comment:
I think you want to do something similar to what `from_longident` does before calling `locate`, i.e. calling `Utils.is_builtin_path`.